### ISSUE TITLE
Caching Interferes with Raw Output in Generated Markdown

### ISSUE THREAD

**Issue by MichaelPascale** [quarto-dev/quarto-cli#7901]
### Bug description

I have a presentation generated with Quarto's [RevealJS](https://quarto.org/docs/presentations/revealjs/#creating-slides) format, and a section of this report contains slides that are generated dynamically in a for loop. 

When I enable [caching](https://quarto.org/docs/computations/r.html#caching) to speed up renders, the content that is intended for separate slides is rendered on top of itself on a single slide.

This behavior appears to be due to the fact that the generated markdown file contains a `.cell` wrapping the generated markdown (to keep track of the source hash), despite having been rendered with `output: asis`.

I don't know whether there's really a way around this (at least I can't think of one myself). Perhaps a documentation note could prevent users from running into this situation.

_P.S. Thank you for Quarto. It's amazing!!_ ❤️

### Steps to reproduce

````
---
format: revealjs
execute:
  cache: false # Toggle this option and compare the output when it is enabled and disabled.
keep-md: true
---

```{r}
#| output: asis
library(ggplot2)

for (i in 1:5) {
  cat('## Slide', i, '\n\n')
  print(ggplot(iris) + geom_point(aes(Sepal.Length, Sepal.Width)))
  cat('\n\n')
}
```
````

### Expected behavior

Generated Markdown:

````
## Slide 1

![](TEST_files/figure-revealjs/unnamed-chunk-1-1.png){width="960"}

## Slide 2

![](TEST_files/figure-revealjs/unnamed-chunk-1-2.png){width="960"}

## Slide 3

![](TEST_files/figure-revealjs/unnamed-chunk-1-3.png){width="960"}

## Slide 4

![](TEST_files/figure-revealjs/unnamed-chunk-1-4.png){width="960"}

## Slide 5

![](TEST_files/figure-revealjs/unnamed-chunk-1-5.png){width="960"}

````

<img width="970" alt="Screen Shot 2023-12-12 at 4 50 56 PM" src="https://github.com/quarto-dev/quarto-cli/assets/32907334/8b210253-c016-462f-bd9c-5ac780551ae8">

<img width="979" alt="Screen Shot 2023-12-12 at 4 51 54 PM" src="https://github.com/quarto-dev/quarto-cli/assets/32907334/ea8af50c-ccb6-4649-8ca3-92b0fa88428b">

### Actual behavior

Generated Markdown:

The wrapping of the code block with the hash for caching purposes interferes with how the slides are rendered.

````
::: {.cell hash='TEST_cache/revealjs/unnamed-chunk-1_28948a0de73705cfb184543f3543f7c1'}
## Slide 1 

![](TEST_files/figure-revealjs/unnamed-chunk-1-1.png){width=960}

## Slide 2 

![](TEST_files/figure-revealjs/unnamed-chunk-1-2.png){width=960}

## Slide 3 

![](TEST_files/figure-revealjs/unnamed-chunk-1-3.png){width=960}

## Slide 4 

![](TEST_files/figure-revealjs/unnamed-chunk-1-4.png){width=960}

## Slide 5 

![](TEST_files/figure-revealjs/unnamed-chunk-1-5.png){width=960}
:::
````

<img width="971" alt="Screen Shot 2023-12-12 at 4 49 58 PM" src="https://github.com/quarto-dev/quarto-cli/assets/32907334/40f24922-f9ad-456e-88a3-c86775ed4391">

### Your environment

**RStudio Server**
RStudio 2023.06.1+524 "Mountain Hydrangea" Release (547dcf861cac0253a8abb52c135e44e02ba407a1, 2023-07-07) for RHEL 8

**Web Browser**
Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:109.0) Gecko/20100101 Firefox/119.0

**Operating System (RStudio Host)**
LSB Version:    :core-4.1-amd64:core-4.1-noarch
Distributor ID: AlmaLinux
Description:    AlmaLinux release 8.9 (Midnight Oncilla)
Release:        8.9
Codename:       MidnightOncilla

### Quarto check output

````
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.1: OK
      Dart Sass version 1.55.0: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 1.3.353
      Path: /usr/local/ood/rstudio-server-2023.06.1-524/bin/quarto/bin

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.10.12
      Path: /share/pkg.8/python3/3.10.12/install/bin/python3
      Jupyter: 5.3.1
      Kernels: python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.2.1
      Path: /share/pkg.7/r/4.2.1/install/lib64/R
      LibPaths:
        - /restricted/projectnb/cd-lab/Michael/human-arc/renv/library/R-4.2/x86_64-pc-linux-gnu
        - /usr3/graduate/mpascale/.cache/R/renv/sandbox/R-4.2/x86_64-pc-linux-gnu/a0c9c547
      knitr: 1.45
      rmarkdown: 2.25

[✓] Checking Knitr engine render......OK
````

**Comment by MichaelPascale** [comment]
Note this appears to be related to #3683

**Comment by cderv** [comment]
This has already been fixed in Quarto 1.4 as part of https://github.com/quarto-dev/quarto-cli/issues/7602 in https://github.com/quarto-dev/quarto-cli/commit/26987f32cf3924f0a5fe151be659b705798b0636 

It was an issue where `hash` should not have been forwarded to the `.cell` div. 

Please to update to latest pre-release to have the fix;  I'll close this as it is fixed.

Side note: 

You need to be ware of how **knitr** cache works. It is not adviced to have library loaded in a cach chunk, because it is side effect (attaching a library in your environment). In your case **ggplot2** would not be available to other chunk for example once cached. 

Also, using `print()` and `cat()` to generate programatically some markdown content could have unexpected behavior. This is because you are not going throught the knitr's printing method for your R objects. It works here, but it may not for all type of object. See in R Markdown cookbook how to generated markdown content from R chunk. 

Some resources about this :
* https://bookdown.org/yihui/rmarkdown-cookbook/results-asis.html
* https://bookdown.org/yihui/rmarkdown-cookbook/knit-expand.html
* https://bookdown.org/yihui/rmarkdown-cookbook/child-document.html

Hope it helps.
